tsfile/src/main/java/org/apache/iotdb/tsfile/read/query/timegenerator/node/AndNode.java

```diff
@@ -41,46 +50,68 @@ public AndNode(Node leftChild, Node rightChild) {
   }
 
   @Override
-  public boolean hasNext() throws IOException {
+  public boolean hasNextTimeColumn() throws IOException {
     if (hasCachedValue) {
       return true;
     }
-    if (leftChild.hasNext() && rightChild.hasNext()) {
-      long leftValue = leftChild.next();
-      long rightValue = rightChild.next();
-      while (true) {
-        if (leftValue == rightValue) {
-          this.hasCachedValue = true;
-          this.cachedValue = leftValue;
-          return true;
-        } else if (leftValue > rightValue) {
-          if (rightChild.hasNext()) {
-            rightValue = rightChild.next();
-          } else {
-            return false;
-          }
-        } else { // leftValue < rightValue
-          if (leftChild.hasNext()) {
-            leftValue = leftChild.next();
-          } else {
-            return false;
-          }
+    cachedTimeColumn = new TimeColumn(1000);
```

Comment:
This magic number can be replaced by a constant.

Reply:
yes ,  maybe can use `fetchSize`